app/views/shared/_language_picker.html.erb

```diff
@@ -0,0 +1,29 @@
+<% sanitized_requested_url = request.query_parameters.slice(:request_id) %>
```

Comment:
For some reason I wasn't initially feeling compelled to make this a ViewComponent instead of a partial, but in retrospect I think that could be a good idea, to make the language picker a bit more portable toward eventual design system inclusion, and to be able to include a preview as part of Lookbook work in #7115.

Reply:
+1 seems like a good component candidate, but not incredibly urgent

```suggestion
<% sanitized_request_params = request.query_parameters.slice(:request_id) %>
```
I know the name is pre-existing but since we're here let's clarify that it's just a params hash?

Reply:
Also if we're not going to make it a component, can we add our pseudo-component locals documentation here?
```erb
<%#
locals
* id_suffix - string to append to element ids so the partial can be used multiple times on a page
%>
```

Reply:
Updated to component in c0357fd. Equivalent method name has been renamed to `sanitized_request_params`.